qubesbuilder/config.py

```diff
@@ -611,6 +613,56 @@ def get_stages(self) -> List[str]:
     def get_plugin_manager(self):
         return PluginManager(self.get_plugins_dirs())
 
+    def get_needs(
+        self,
+        component: QubesComponent,
+        dist: QubesDistribution,
+        stage_name: str,
+    ):
+        needs = []
+        stages = component.kwargs.get(dist.distribution, {}).get("stages", [])
+        for stage in stages:
+            if (
+                isinstance(stage, dict)
+                and next(iter(stage)) == stage_name
+                and isinstance(stage[stage_name], dict)
+            ):
+                for need in stage[stage_name].get("needs", []):
+                    if all(
+                        [
+                            need.get("component", None),
+                            need.get("distribution", None),
+                            need.get("stage", None),
+                            need.get("build", None),
+                        ]
```

Comment:
and same here - if some part is missing, at least log a warning why it's ignored